wayc: updates by gurjaka

Gurjaka opened the pull request.

Gurjaka committed:
wayc: fix the build, adapt wlroots 0.19

Gurjaka committed:
wayc: add descriptive, and concise comments

Gurjaka committed:
wayc: git ignore ccls cache

jwijenbergh commented on .gitignore:
```diff
@@ -58,3 +58,10 @@ docs/_static/screenshots/
 /tags
+
+# nix direnv
+.envrc
```
These are unrelated to wayc can we drop this and put the pr in main qtile repo?

Gurjaka replied:
sure, it just helped me ignore .envrc and .direnv instead of removing them each time I make a commit, also I made the pr here, so you could merge to your wayc, and make pr afterwards, I don't really know how it works if I make pr on main Qtile repo, could we discuss further in discord?

dannycolin replied:
You can add them to your global git ignore.

Gurjaka replied:
true.

jwijenbergh commented on pyproject.toml:
```diff
@@ -13,10 +13,9 @@ build-backend = "builder"
 name = "qtile"
 description = "A pure-Python tiling window manager."
 dynamic = ["version", "readme", "dependencies", "optional-dependencies"]
-license = {text = "MIT"}
```
These are unrelated to wayc can we drop this and put the pr in main qtile repo?

Gurjaka replied:
this isn't related to Wayc, but since the fork wasn't synced, I had to make manual change to this in order to get qtile to build

jwijenbergh commented on libqtile/backend/wayland/qw/view.h:
```diff
@@ -4,51 +4,58 @@
-    TOP = 5,
-    MINIMIZED = 6,
+    NOT_FLOATING = 1, // normal tiled window
+    FLOATING = 2,     // floating window
```
Like e.g. this comment, not very useful

jwijenbergh commented on libqtile/backend/wayland/qw/xdg-view.c:
```diff
@@ -108,7 +108,7 @@ static void qw_xdg_view_place(void *self, int x, int y, int width, int height, i
                          state.width != width || state.height != height;
     struct wlr_box geom;
-    wlr_xdg_surface_get_geometry(surface, &geom);
+    wlr_surface_get_extents(surface->surface, &geom);
```
this should be `struct wlr_box geom = surface->geometry` instead

Gurjaka replied:
okay will fix rn

Gurjaka replied:
done

Gurjaka commented:
accidentally merged the comments commit with previous one :cry:  will fix asap

Gurjaka committed:
wayc: fix wlr_box geom

Gurjaka commented:
> accidentally merged the comments commit with previous one 😢 will fix asap

okay it should be fixed now, with applied `wlr_box` fix as well

jwijenbergh commented:
Done most in: ﻿﻿https://github.com/qtile/qtile/commit/2ad79a011c7c4bcdce37e0e04c5c5f8d4389ebf6

Gurjaka commented:
> Done most in: [qtile@3268e83](https://github.com/qtile/qtile/commit/3268e833ac499123805be313246bb77b4824a91d)

Perfect, now all we need is to sync the branch and open issues

jwijenbergh commented:
> > Done most in: [qtile@3268e83](https://github.com/qtile/qtile/commit/3268e833ac499123805be313246bb77b4824a91d)
> 
> Perfect, now all we need is to sync the branch and open issues

Will do this later!

Gurjaka committed:
wayc: fix the Cairo surface pointer issue

The TypeError occurs because of a pointer type mismatch between Cairo and wlroots.
Solution: First, modify internal-view.h to expose the proper surface
type, then update drawer.py

Gurjaka commented:
I also fixed following error:
```bash
  File "/nix/store/nqvgxmg52894mvw5gv95hnsdi82zrmkw-python3.12-cairocffi-1.7.1/lib/python3.12/site-packages/cairocffi/surfaces.py", line 187, in _from_pointer
    cairo.cairo_surface_reference(pointer)
TypeError: initializer for ctype 'cairo_surface_t *' must be a pointer to same type, not cdata 'struct _cairo_surface *'
```

didn't make duplicate of this PR on official Qtile, will just leave merging to you :)

Gurjaka commented:
> I also fixed following error:
> 
> ```shell
>   File "/nix/store/nqvgxmg52894mvw5gv95hnsdi82zrmkw-python3.12-cairocffi-1.7.1/lib/python3.12/site-packages/cairocffi/surfaces.py", line 187, in _from_pointer
>     cairo.cairo_surface_reference(pointer)
> TypeError: initializer for ctype 'cairo_surface_t *' must be a pointer to same type, not cdata 'struct _cairo_surface *'
> ```
> 
> didn't make duplicate of this PR on official Qtile, will just leave merging to you :)

oh I just checked the Qtile wayc, looks like you already fixed it other way, but still I'll leave it here, maybe this is more acceptable?

Gurjaka commented:
also cleaned up some noisy comments, for now will just wait for your review
